docs/release.md

```diff
@@ -0,0 +1,180 @@
+# OSA release process
+
+OSA is designed to allow customers to choose an update `stream` for their cluster. Currently the only stream is `3.11`, but this may change over time. A single release of the OSA project can in principle support co-existent clusters at multiple streams. In principle, clusters will one day be upgradable between streams. This is not in scope for this document.
```

Comment:
nit: please wrap throughout, always

Reply:
Can you elaborate on this?

Reply:
break up long lines in <80c long lines, I think is what he meant.